.ci/lib.sh

```diff
@@ -80,7 +80,7 @@ function build_version() {
 
 	if [ "$version" != "HEAD" ]; then
 		info "Using ${github_project} version ${version}"
-		git checkout -b "${version}" "${version}"
```

Comment:
LGTM.

What about making the branche name something transient, e.g. with a date?  `test-$(date +%Y-%m-%d-%H:%M:%S)-${version}`?

Reply:
We could, but why bother?  I don't actually see any real benefit to creating the branch, and it pollutes the other tree with a bunch of stale branch tags.

Reply:
Sorry, I was not awake enough, read the change backwards.

I assume `git checkout` should always succeed, possibly creating a local branch if you have some origin version here.

Reply:
I don't quite follow.  git checkout will only create a branch with -b, but it will still succeed without creating a branch (putting the tree in detached HEAD state).

Reply:
If you give a sha1, then yes, `git checkout` will always succeed. If you pass a branch, it needs the branch to exist somewhere it can be tracked from. But I guess that is always the case here.

Reply:
Right, yes.  Of the examples I've seen, one uses a sha, one uses a tag.  A existing branch name would not previously have worked, since it would have tried to create a new branch with the same name as the existing one.  So, as a bonus the change should allow for versions to be set to existing branches.  That's probably not a wise choice, though, since the point here is to get a specific version to build.

Reply:
It's worth pointing out that this code isn't only run by the CI - if a developer runs https://github.com/kata-containers/tests/blob/master/.ci/static-checks.sh, that can call `build_version()` so I wonder if it would less surprising to those users if we changed this to use `-B` (aka "switch to branch and reset, but don't fail if branch already exists"):

```
git checkout -B "${version}" "${version}"
```

/cc @jcvenegas, @chavafg, @amshinde.

Reply:
Least surprising is a bit hard to judge.  But simply moving the other tree to a detached HEAD state seems less dangerous to me than adding or resetting a branch there.

That said, either one addresses the initial problem here.

Reply:
That's a good point - let's go with this.